### PR TITLE
feat: Add WorkQueueClient to the provider

### PR DESCRIPTION
**Pull request by jamiezieziula**

### CODE REVIEW

**Review by gabcoyne — APPROVED** [review]
👍🏻

File: internal/client/work_queues.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	if accountID == uuid.Nil {
+		accountID = c.defaultAccountID
+	}
+	if workspaceID == uuid.Nil {
+		workspaceID = c.defaultWorkspaceID
+	}
+
+	var workspaceScopedURL = getWorkspaceScopedURL(c.endpoint, accountID, workspaceID, "work_pools")
+
+	var workPoolScopedURL = workspaceScopedURL + "/" + workPoolName + "/queues"
```
Comment (parkedwards) [inline internal/client/work_queues.go:37]: not opposed to this, but that means one of these `WorkQueue` client instances will be tied to a specific workPool

File: internal/client/work_queues.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+	setDefaultHeaders(req, c.apiKey)
+
+	resp, err := c.hc.Do(req)
+	if err != nil {
+		return fmt.Errorf("http error: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
```
Comment (parkedwards) [inline internal/client/work_queues.go:163]: just making sure - is 200 and 204 both a valid response code for PATCH /queue?

Comment (jamiezieziula) [reply internal/client/work_queues.go]: updated!

**Review by parkedwards — APPROVED** [review]
🥳

**Pull request merged**